mqtt-v5/src/types.rs

```diff
@@ -34,6 +34,12 @@ pub enum DecodeError {
     BadTransport, // When errors occur on a lower level transport like WS
 }
 
+impl PartialEq for DecodeError {
+    fn eq(&self, other: &Self) -> bool {
+        self == other
```

Comment:
I read this and thought "Isn't this recursive?"

Indeed it is, since the `==` operator uses `PartialEq`, so you can't define `PartialEq` in terms of itself.